zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/UserTaskHandlerTest.java

```diff
@@ -512,15 +585,16 @@ void flushedEntityShouldContainUpdates() {
     verify(mockRequest, times(1))
         .upsertWithRouting(
             indexName,
-            taskEntity.getId(),
+            recordKey < firstSeenKey
+                ? String.valueOf(recordKey)
+                : String.valueOf(flowNodeInstanceKey),
```

Comment:
🔧 This is a good sign of splitting a test to scope them for there different use cases